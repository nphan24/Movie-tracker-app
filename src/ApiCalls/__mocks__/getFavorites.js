export const getFavorites = jest.fn().mockImplementation(() => {
  return Promise.resolve({ data: [{ movie: 'movie' }] });
});
