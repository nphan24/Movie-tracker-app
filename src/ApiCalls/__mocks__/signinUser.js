export const signinUser = jest.fn().mockImplementation(() => {
  return Promise.resolve({ id: 2 });
});
