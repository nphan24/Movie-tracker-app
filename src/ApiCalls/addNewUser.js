export const addNewUser = async ({ name, email, password }) => {
  try {
    const response = await fetch("/api/users/new", {
      method: "POST",
      body: JSON.stringify({
        name: name,
        email: email,
        password: password
      }),
      headers: {
        "Content-Type": "application/json"
      }
    });
    const data = await response.json();
    if (data.error) {
      throw new Error(data.error);
    } else {
      return data.id;
    }
  } catch (error) {
    throw new Error(`${email} already exists. Please enter a new email`);
  }
};
