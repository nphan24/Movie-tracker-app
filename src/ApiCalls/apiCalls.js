import apiKey from './apiKey';

export const fetchData = async () => {
  try {
  const response = await fetch(
    `https://api.themoviedb.org/3/movie/upcoming?api_key=${apiKey}`
  );
  const data = await response.json();
  return data.results
  } catch (error) {
    throw 'error'
  }
} 

export const getMovies = async () => {
  const movieData = await fetchData();
  console.log(movieData);
  // const cleanedMovies = cleanMovies(movieData);
  // return cleanedMovies
}

export const cleanMovies = (movieData) => {
  const cleanedMovies = movieData.map(movie => {

  })
}
 
