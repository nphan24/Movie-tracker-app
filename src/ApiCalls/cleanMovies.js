export const cleanMovies = movieData => {
  const cleanedMovies = movieData.map(movie => {
    const {
      title,
      overview,
      release_date,
      vote_average,
      poster_path,
      popularity
    } = movie;
    return {
      title,
      movie_id: Math.floor(Math.random() * 100000),
      overview: `Summary: ${overview}`,
      release_date: `Release Date: ${release_date}`,
      vote_average: `Rating: ${vote_average}`,
      poster_path: poster_path
    };
  });
  return cleanedMovies;
};
