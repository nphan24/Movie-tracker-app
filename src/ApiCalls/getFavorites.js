export const getFavorites = async id => {
  console.log('welcome to getFavs');

  try {
    console.log('TRY in to getFavs');
    const response = await fetch(`api/users/${id}/favorites`);
    const favoriteMovies = await response.json();
    return favoriteMovies;
  } catch (error) {
    console.log('CATCH to getFavs');
    throw new Error(error.message);
  }
};
