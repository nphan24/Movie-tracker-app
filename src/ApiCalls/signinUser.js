export const signinUser = async () => {
  console.log('ADDING!');

  try {
    const response = await fetch('api/users', {
      method: 'POST',
      body: JSON.stringify({email: "tman2272@aol.com", password: "password"}), 
      headers: {
        'Content-Type': 'application/json'
      }
    });
    const data = await response.json();
    return data;
  } catch (errs) {
    console.log(errs.message);
  }
};
