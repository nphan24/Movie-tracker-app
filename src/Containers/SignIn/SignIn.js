import React, { Component } from 'react';
import { connect } from 'react-redux';
import { signinUser } from '../../ApiCalls/signinUser';
import { NavLink, withRouter } from 'react-router-dom';
import {
  signInAction,
  invalidSignIn,
  validSignIn,
  addExistingFavs
} from '../../Actions';
import { getFavorites } from '../../ApiCalls/getFavorites';
import PropTypes from 'prop-types';

export class SignIn extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleInput = event => {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  };

  handleSubmit = async () => {
    try {
      const user = await signinUser({ ...this.state });
      this.props.signInDispatch(user);
      this.props.validSignIn(false);
      const existingFavs = await getFavorites(user.id);
      this.props.addExistingFavs(existingFavs.data);
    } catch (error) {
      this.props.invalidSignIn(error.message);
    }
  };

  render() {
    return (
      <form>
        <NavLink to="/new-user">Sign Up</NavLink>
        <label>
          <input
            type="email"
            placeholder="Email"
            value={this.state.email}
            name="email"
            onChange={this.handleInput}
          />
        </label>
        <label>
          <input
            type="password"
            placeholder="Password"
            onChange={this.handleInput}
            name="password"
            value={this.state.password}
          />
        </label>
        <NavLink to="/" onClick={this.handleSubmit}>
          <button>Submit</button>
        </NavLink>
      </form>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  signInDispatch: user => dispatch(signInAction(user)),
  invalidSignIn: error => dispatch(invalidSignIn(error)),
  validSignIn: valid => dispatch(validSignIn(valid)),
  addExistingFavs: existingFavs => dispatch(addExistingFavs(existingFavs))
});

SignIn.propTypes = {
  signInDispatch: PropTypes.func,
  invalidSignIn: PropTypes.func,
  validSignIn: PropTypes.func,
  promptSignin: PropTypes.func,
  addExistingFavs: PropTypes.func
};

export default withRouter(connect(null, mapDispatchToProps)(SignIn));
