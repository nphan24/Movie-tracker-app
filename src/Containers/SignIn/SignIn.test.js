import React from 'react';
import { SignIn } from './SignIn';
import { shallow } from 'enzyme';
import { signinUser } from '../../ApiCalls/signinUser';
import { getFavorites } from '../../ApiCalls/getFavorites';
jest.mock('../../ApiCalls/signinUser.js');
jest.mock('../../ApiCalls/getFavorites.js');

describe('Login', () => {
  let wrapper;
  let mockSignInDispatch;
  let mockValidSignIn;
  let mockInvalidSignIn;
  let mockAddExistingFavs;
  beforeEach(() => {
    mockSignInDispatch = jest.fn();
    mockValidSignIn = jest.fn();
    mockInvalidSignIn = jest.fn();
    mockAddExistingFavs = jest.fn();
    wrapper = shallow(
      <SignIn
        signInDispatch={mockSignInDispatch}
        validSignIn={mockValidSignIn}
        invalidSignIn={mockInvalidSignIn}
        addExistingFavs={mockAddExistingFavs}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on input change', () => {
    const mockEvent = { target: { value: 'ahhhhh', name: 'name' } };
    wrapper.instance().handleInput(mockEvent);
    expect(wrapper.state('name')).toEqual('ahhhhh');
  });

  it('should sign in a user', async () => {
    await wrapper.instance().handleSubmit();
    expect(signinUser).toHaveBeenCalled();
  });

  it('should call signInDispatch', async () => {
    await wrapper.instance().handleSubmit();
    expect(mockSignInDispatch).toHaveBeenCalled();
  });

  it('should call validSignIn', async () => {
    await wrapper.instance().handleSubmit();
    expect(mockValidSignIn).toHaveBeenCalled();
  });

  it('should run getFavorites fetch', async () => {
    await wrapper.instance().handleSubmit();
    await expect(getFavorites).toHaveBeenCalled();
  });

  it('should call addExistingFavs', async () => {
    await wrapper.instance().handleSubmit();
    expect(mockAddExistingFavs).toHaveBeenCalled();
  });

  it('should run invalidSignIn on error', async () => {
    mockSignInDispatch = jest.fn().mockImplementation(() => {
      throw new Error('Error');
    });
    wrapper = shallow(
      <SignIn
        signInDispatch={mockSignInDispatch}
        invalidSignIn={mockInvalidSignIn}
      />
    );
    await wrapper.instance().handleSubmit();
    expect(mockInvalidSignIn).toHaveBeenCalled();
  });
});
