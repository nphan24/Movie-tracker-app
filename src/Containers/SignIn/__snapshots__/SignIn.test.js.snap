// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    addExistingFavs={[Function]}
    invalidSignIn={[Function]}
    signInDispatch={[Function]}
    validSignIn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/new-user"
>
          Sign Up
</NavLink>,
        <label>
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
                    value=""
          />
</label>,
        <label>
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
</label>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          onClick={[Function]}
          to="/"
>
          <button>
                    Submit
          </button>
</NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Sign Up",
          "to": "/new-user",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button>
            Submit
</button>,
          "onClick": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/new-user"
>
            Sign Up
</NavLink>,
          <label>
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="email"
                        value=""
            />
</label>,
          <label>
            <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
</label>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            onClick={[Function]}
            to="/"
>
            <button>
                        Submit
            </button>
</NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Sign Up",
            "to": "/new-user",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button>
              Submit
</button>,
            "onClick": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
